tools/regression_test.py

```diff
@@ -0,0 +1,987 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import argparse
+import logging
+import os
+from collections import OrderedDict
+from pathlib import Path
+
+import mmcv
+import pandas as pd
+import yaml
+from torch.hub import download_url_to_file
+from torch.multiprocessing import set_start_method
+
+from mmdeploy.utils import (get_backend, get_codebase, get_root_logger,
+                            is_dynamic_shape, load_config)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description='Regression Test')
+    parser.add_argument(
+        '--deploy-yml',
+        nargs='+',
+        help='regression test yaml path.',
+        default=[
+            './configs/mmdet/mmdet_regression_test.yaml',
+            './configs/mmcls/mmcls_regression_test.yaml',
+            './configs/mmseg/mmseg_regression_test.yaml',
+            './configs/mmpose/mmpose_regression_test.yaml',
+            './configs/mmocr/mmocr_regression_test.yaml',
+            './configs/mmedit/mmedit_regression_test.yaml'
+        ])
+    parser.add_argument(
+        '--performance',
+        default=False,
+        action='store_true',
+        help='test performance if it set')
+    parser.add_argument(
+        '--backends',
+        nargs='+',
+        help='test specific backend(s)',
+        default=['all'])
+    parser.add_argument(
+        '--work-dir',
+        type=str,
+        help='the dir to save logs and models',
+        default='../mmdeploy_regression_working_dir')
+    parser.add_argument(
+        '--device', type=str, help='`the CUDA device id', default='cuda')
+    parser.add_argument(
+        '--log-level',
+        help='set log level',
+        default='INFO',
+        choices=list(logging._nameToLevel.keys()))
+    args = parser.parse_args()
+
+    return args
+
+
+def get_model_metafile_info(global_info, model_info, logger):
+    """Get model metafile information.
+
+    Args:
+        global_info (dict): global info from deploy yaml.
+        model_info (dict):  model info from deploy yaml.
+        logger (logging.Logger): logger.
+
+    Returns:
+        Dict: Meta info of each model config
+    """
+
+    # get info from global_info and model_info
+    checkpoint_dir = global_info.get('checkpoint_dir', None)
```

Comment:
could we use the default checkpoint cache directory of the system as the default choice of `checkpoint_dir`?

Reply:
OK, See if I get it right : 
If user set `None` to `checkpoint_dir` , I will use the default checkpoint cache directory of the system ?

Another question: Where is `default checkpoint cache directory of the system`, is it mmlab have one, or just download in the path which decide by the system ?

Reply:
Actually, it is tools/deploy.py that accepts checkpoint URL as the weight input. It is decided by codebases like mmdet where to save the downloaded checkpoint. Basically, the checkpoints are saved into ~/.cache/...

Reply:
After consideration, I think we need to use `checkpoint_dir`, cuz after the regression test, it will be reports for each codebase and some convert info for each backend. Also, using `checkpoint_dir` we can access those files or reports easily and we can clean all of it easily at the same time.